### DISCUSSION TITLE
TestClient hangs on synchronous endpoints

### DISCUSSION THREAD

**Discussion by AvivSamet-Silk** [fastapi/fastapi#5979] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
"""
CONFTEST.PY
"""

import asyncio
import pytest
from typing import Iterator

from fastapi.testclient import TestClient

from my_app import app


@pytest.fixture(scope="module")
def event_loop() -> Iterator[asyncio.AbstractEventLoop]:
    loop = asyncio.get_event_loop_policy().new_event_loop()
    yield loop
    loop.close()


@pytest.fixture(scope="module")
async def api_server():
    with TestClient(app=app) as test_server:
        yield test_server

"""
TEST_ENDPOINTS.PY
"""

def test_post_endpoint(self, api_server: TestClient):

        response = api_server.post("/example")

        assert response.status_code == status.HTTP_200_OK

def test_delete_endpoint(self, api_server: TestClient):

        response = api_server.delete("/example")

        assert response.status_code == status.HTTP_200_OK

"""
MY_APP.PY
"""

@router.post("/example", status_code=status.HTTP_200_OK)
def post_endpoint(customer: Customers, create_req: ProjectCreateReq):

    return {"status": "ok"}

@router.delete("/example", status_code=status.HTTP_200_OK)
def delete_endpoint(customer: Customers, create_req: ProjectCreateReq):

    return {"status": "ok"}
```

### Description

I initially built a test for endpoints that were async, but once the endpoints were converted to synchronous the tests started hanging every time that the tests finished. I tried several things, including creating a new event loop as was suggested in [other issues](https://github.com/tiangolo/fastapi/issues/5692). Creating a new event loop doesn't stop the test from hanging, and trying to call `api_server.close()` gives me the following error: 
![image](https://user-images.githubusercontent.com/108717750/217657436-1e180f8e-c9fa-49b1-8d54-643868647574.png)

I found some similar problems in [this thread](https://github.com/python-hyper/hyper/issues/306), but in this case they actually did an override of the close in the client.

### Operating System

Linux

### FastAPI Version

0.75.2

### Python Version

3.10.6

**Comment by jkehrbaum** [comment, marked as answer]
Try using a proper async TestClient such as AsyncClient from httpx. You are mixing a non async client with an async fixture, which ends up in a mess …

Cheers Jan
